app/controllers/vendor_api/decisions_controller.rb

```diff
@@ -51,6 +52,17 @@ def respond_to_decision(decision)
       end
     end
 
+    def offered_course_option
+      course_data = params.dig(:data, :course)
+      return nil if course_data.nil?
+
+      provider = Provider.find_by(code: course_data[:provider_code])
+      course = provider.courses.find_by(code: course_data[:course_code])
+      site = provider.sites.find_by(code: course_data[:site_code])
```

Comment:
This should potentially search by `study_mode` as well, depending on if https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/851 is merged.